base/v0_5_exp/translate_test.go

```diff
@@ -1744,6 +1744,292 @@ func TestTranslateKernelArguments(t *testing.T) {
+					"$.sshAuthorizedKeys.4": {From: path.ContextPath{Path: []interface{}{"ssh_authorized_keys_local"}, Tag: "yaml"}, To: path.ContextPath{Path: []interface{}{"sshAuthorizedKeys", 4}, Tag: "json"}},
+					"$.sshAuthorizedKeys.5": {From: path.ContextPath{Path: []interface{}{"ssh_authorized_keys_local"}, Tag: "yaml"}, To: path.ContextPath{Path: []interface{}{"sshAuthorizedKeys", 5}, Tag: "json"}},
+					"$.sshAuthorizedKeys.6": {From: path.ContextPath{Path: []interface{}{"ssh_authorized_keys_local"}, Tag: "yaml"}, To: path.ContextPath{Path: []interface{}{"sshAuthorizedKeys", 6}, Tag: "json"}},
+				},
+			},
+			"",
+			sshKeyDir,
+			"",
+		},
+
+		{
+			"valid empty ssh_keys_local file",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyEmptyFileName}},
+			types.PasswdUser{SSHAuthorizedKeys: []types.SSHAuthorizedKey{types.SSHAuthorizedKey("")}},
+			translate.TranslationSet{
+				FromTag: "yaml",
+				ToTag:   "json",
+				Set: map[string]translate.Translation{
+					"$.name":                {From: path.ContextPath{Path: []interface{}{"name"}, Tag: "yaml"}, To: path.ContextPath{Path: []interface{}{"name"}, Tag: "json"}},
+					"$.sshAuthorizedKeys":   {From: path.ContextPath{Path: []interface{}{"ssh_authorized_keys_local"}, Tag: "yaml"}, To: path.ContextPath{Path: []interface{}{"sshAuthorizedKeys"}, Tag: "json"}},
+					"$.sshAuthorizedKeys.0": {From: path.ContextPath{Path: []interface{}{"ssh_authorized_keys_local"}, Tag: "yaml"}, To: path.ContextPath{Path: []interface{}{"sshAuthorizedKeys", 0}, Tag: "json"}},
+				},
+			},
+			"",
+			sshKeyDir,
+			"",
+		},
+		{
+			"valid blank ssh_keys_local file",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyBlankFileName}},
+			types.PasswdUser{SSHAuthorizedKeys: []types.SSHAuthorizedKey{types.SSHAuthorizedKey(""), types.SSHAuthorizedKey("\t")}},
+			translate.TranslationSet{
+				FromTag: "yaml",
+				ToTag:   "json",
+				Set: map[string]translate.Translation{
+					"$.name":                {From: path.ContextPath{Path: []interface{}{"name"}, Tag: "yaml"}, To: path.ContextPath{Path: []interface{}{"name"}, Tag: "json"}},
+					"$.sshAuthorizedKeys":   {From: path.ContextPath{Path: []interface{}{"ssh_authorized_keys_local"}, Tag: "yaml"}, To: path.ContextPath{Path: []interface{}{"sshAuthorizedKeys"}, Tag: "json"}},
+					"$.sshAuthorizedKeys.0": {From: path.ContextPath{Path: []interface{}{"ssh_authorized_keys_local"}, Tag: "yaml"}, To: path.ContextPath{Path: []interface{}{"sshAuthorizedKeys", 0}, Tag: "json"}},
+					"$.sshAuthorizedKeys.1": {From: path.ContextPath{Path: []interface{}{"ssh_authorized_keys_local"}, Tag: "yaml"}, To: path.ContextPath{Path: []interface{}{"sshAuthorizedKeys", 1}, Tag: "json"}},
+				},
+			},
+			"",
+			sshKeyDir,
+			"",
+		},
+		{
+			"valid Windows style line endings in ssh_keys_local file", // TODO fix this; Windows line endings are a problem on Linux & macOS :/
```

Comment:
Curse me for not writing a better comment – what I was trying to say was roughly the following (if memory serves):

Linux & macOS builds (or test execution?) was having trouble printing/parsing Windows-style line endings. There was some very weird edge case going on. I think some empty keys were accidentally emitted because of the way line splitting was previously handled, i.e. for each `\r` an additional blank key was injected into the authorized keys list; Windows was fine, though, because it handles `\n` and `\r\n` the same.

Looks fine, though (🤞), but I'll have another look with a fresh head.